database/src/main/java/database/service/SqlDatabaseService.java

```diff
@@ -450,6 +537,56 @@ private void setPreparedStatementValues(PreparedStatement preparedStatement, Bas
         }
     }
 
+    private <T extends BaseEntity> void setPreparedStatementValues(PreparedStatement preparedStatement,
+                                                                   List<T> entities, boolean includeId,
+                                                                   List<Integer> ids) {
+        int parameterIndex = 1;
+
+        try {
+            for (int entityIndex = 0; entityIndex < entities.size(); entityIndex++) {
+                T entity = entities.get(entityIndex);
+                List<Field> entityFields = getAllFields(entity.getClass());
+
+                for (Field field : entityFields) {
+                    if (field.getName().equals(ID_PARAMETER_NAME)) {
+                        continue;
+                    }
+
+                    field.setAccessible(true);
+                    Object value = field.get(entity);
+
+                    if (value != null) {
```

Comment:
There are a lot of common code here and in `setPreparedStatementValues` method, consider moving common code to separate method.